pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -1892,3 +1892,213 @@ func createChainedEcOpHinter(resolver hintReferenceResolver) (hinter.Hinter, err
 
 	return newChainedEcOpHint(len, p, m, q, s), nil
 }
+
+// BigIntPackDivMod hint divides two values modulo a prime number
+//
+// `newBigIntPackDivModHint` takes 3 operanders as arguments
+//   - `P` is the prime modulus
+//   - `x` is the numerator
+//   - `y` is the denominator
+//
+// `newBigIntPackDivModHint` assigns the result as `value` in the current scope
+func newBigIntPackDivModHint(x, y, p hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivMod",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import pack
+			//> from starkware.cairo.common.math_utils import as_int
+			//> from starkware.python.math_utils import div_mod, safe_div
+			//>
+			//> p = pack(ids.P, PRIME)
+			//> x = pack(ids.x, PRIME) + as_int(ids.x.d3, PRIME) * ids.BASE ** 3 + as_int(ids.x.d4, PRIME) * ids.BASE ** 4
+			//> y = pack(ids.y, PRIME)
+			//>
+			//> value = res = div_mod(x, y, p)
+
+			pAddr, err := p.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			pValues, err := vm.Memory.ResolveAsBigInt3(pAddr)
+			if err != nil {
+				return err
+			}
+
+			xAddr, err := x.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			xValues, err := vm.Memory.ResolveAsBigInt3(xAddr)
+			if err != nil {
+				return err
+			}
+
+			yAddr, err := y.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			yValues, err := vm.Memory.ResolveAsBigInt3(yAddr)
+			if err != nil {
+				return err
+			}
+
+			var xD3Big big.Int
+			var xD4Big big.Int
+
+			xValues[0].BigInt(&xD3Big)
+			xValues[1].BigInt(&xD4Big)
+
+			base, ok := secp_utils.GetBaseBig()
+			if !ok {
+				return fmt.Errorf("getBaseBig failed")
+			}
+
+			//> p = pack(ids.P, PRIME)
+			pBig, err := secp_utils.SecPPacked(pValues)
+			if err != nil {
+				return err
+			}
+
+			//> x = pack(ids.x, PRIME) + as_int(ids.x.d3, PRIME) * ids.BASE ** 3 + as_int(ids.x.d4, PRIME) * ids.BASE ** 4
+			xPacked, err := secp_utils.SecPPacked(xValues)
+			if err != nil {
+				return err
+			}
+
+			base3Big := new(big.Int)
+
+			base3Big.Exp(&base, big.NewInt(3), big.NewInt(0))
+
+			base4Big := new(big.Int)
+
+			base4Big.Exp(&base, big.NewInt(4), big.NewInt(0))
+
+			xBig := new(big.Int)
+
+			xBig.Mul(&xD3Big, base3Big)
+
+			xBig.Add(xBig, xBig.Mul(&xD4Big, base4Big))
+
+			xBig.Add(xBig, &xPacked)
+
+			//> y = pack(ids.y, PRIME)
+			yBig, err := secp_utils.SecPPacked(yValues)
+			if err != nil {
+				return err
+			}
+
+			//> value = res = div_mod(x, y, p)
+			res, err := secp_utils.Divmod(xBig, &yBig, &pBig)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(xBig)
+			fmt.Println(&yBig)
+			fmt.Println(&pBig)
+
+			return ctx.ScopeManager.AssignVariables(map[string]any{"value": &res})
```

Comment:
note that you also need to assign `res` in scope here, because the next hint retrieves `res` from the scope